cypress/integration/patch/patch-route/tasks-filters.js

```diff
@@ -36,8 +37,102 @@ describe("Tasks filters", function() {
       locationHasUpdatedFilterParam(null);
     });
   });
+
+  describe("Statuses select", () => {
+    beforeEach(() => {
+      cy.get("[data-cy=task-status-filter] > .cy-treeselect-bar").click();
+    });
+
+    it("Clicking on 'All' checkbox adds all statuses to URL. Clicking again removes all statuses.", () => {
+      cy.get(".cy-checkbox")
+        .contains("All")
+        .then($all => {
+          $all.click();
+          locationHasUpdatedFilterParam(allStatuses, "statuses");
+          $all.click();
+          locationHasUpdatedFilterParam(null);
+        });
+    });
+
+    [
+      {
+        title: "Failures",
+        statuses: "all-failures,failed,test-timed-out"
+      },
+      {
+        title: "Scheduled",
+        statuses: "scheduled,unstarted,undispatched"
+      },
+      {
+        title: "System Issues",
+        statuses: "system-issues,system-failed"
+      }
+    ].forEach(({ title, statuses }) => {
+      it(`Clicking on a parent selector '${title}' updates url status param with it and all its children`, () => {
+        clickCheckboxesAndAssertCorrectUrlParams(title, statuses);
+      });
+    });
+
+    [
+      {
+        title: "Failed",
+        key: "failed"
+      },
+      {
+        title: "Test Timed Out",
+        key: "test-timed-out"
+      },
+      {
+        title: "Success",
+        key: "success"
+      },
+      {
+        title: "Running",
+        key: "dispatched"
+      },
+      {
+        title: "Started",
+        key: "started"
+      },
+      {
+        title: "Unstarted",
+        key: "unstarted"
+      },
+      {
+        title: "Undispatched",
+        key: "undispatched"
+      },
+      {
+        title: "System Failed",
+        key: "system-failed"
+      },
+      {
+        title: "Setup Failed",
+        key: "setup-failed"
+      },
+      {
+        title: "Blocked",
+        key: "blocked"
+      }
+    ].forEach(({ title, key }) => {
+      it(`Clicking on singular statuses '${title}' updates url status with '${key}'`, () => {
```

Comment:
It's worth checking that ${key}, does not exist so we know there aren't many additional statuses seperated by a comma